### ISSUE TITLE
[BUG] NPE in XQTS `fn-base-uri` tests

### ISSUE THREAD

**Issue by dizzzz** [eXist-db/exist#3699]
XQTS shows the following NPE in `fn-base-uri` tests

```
Caused by: java.lang.NullPointerException
at org.exist.dom.memtree.DocumentImpl.getBaseURI(DocumentImpl.java:1476)
at org.exist.xquery.functions.fn.FunBaseURI.eval(FunBaseURI.java:163)
at org.exist.xquery.BasicFunction.eval(BasicFunction.java:73)
at org.exist.xquery.InternalFunctionCall.eval(InternalFunctionCall.java:62)
at org.exist.xquery.PathExpr.eval(PathExpr.java:271)
at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:82)
at org.exist.xquery.ForExpr.eval(ForExpr.java:128)
at org.exist.xquery.PathExpr.eval(PathExpr.java:271)
at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
at org.exist.xquery.XQuery.execute(XQuery.java:261)
at org.exist.xquery.XQuery.execute(XQuery.java:185)
at org.exist.xquery.XQuery.execute(XQuery.java:180)
at org.exist.xqts.runner.ExistConnection.$anonfun$executeQuery$36(ExistServer.scala:306)
at flatMap @ org.exist.xqts.runner.ExistConnection.$anonfun$executeQuery$35(ExistServer.scala:313)
at flatMap @ org.exist.xqts.runner.ExistConnection.$anonfun$executeQuery$33(ExistServer.scala:303)
at unsafeRunSync @ org.exist.xqts.runner.Checksum$.checksum(Checksum.scala:81)
at make @ org.exist.xqts.runner.Checksum$.checksum(Checksum.scala:70)
at make @ org.exist.xqts.runner.Checksum$.checksum(Checksum.scala:70)
at use @ org.exist.xqts.runner.Checksum$.checksum(Checksum.scala:70)
```

shown with

```xquery
xquery version "3.1";
declare context item := <a>b</a>;

for $i in (1, base-uri(.), 3) return 
        typeswitch($i) 
        case xs:anyURI return "xs:anyURI" 
        case xs:integer return "xs:integer" 
        default return "FAILURE"
```

tests:  eg `fn-base-uri | K2-BaseURIFunc-21`

**Comment by adamretter** [comment]
@dizzzz As the route of the error is in `org.exist.dom.memtree.DocumentImpl`, it you make any changes to that class, it would be sensible to run the [DOM Test Suite](https://github.com/eXist-db/exist-domts) before and after just to avoid any regressions.

Unfortunately as eXist-db implements DOM and XDM in the same classes, it can be sometimes quite tricky to find a solution that works for bith.

**Comment by dizzzz** [comment]
I'd say to see some improvement?

<img width="1377" alt="image" src="https://user-images.githubusercontent.com/1700062/104133579-b9078b80-5384-11eb-9243-f4d459d6ad25.png">

**Comment by adamretter** [comment]
@dizzzz If I understand your dom-ts results correctly, then we see a difference of -4 errors and +3 failures. So an improvement of 1 overall. Which is good of course :-) I just wondered if you knew why 3 previous errors are now registered as failures?

**Comment by dizzzz** [comment]
In XQTS the fixes turn some tests into OK (progression!) and a number of exceptions (NPEs) into test failures.
I assume the same is valid here.

**Comment by joewiz** [comment]
@dizzzz This was closed by #3813, right?

**Comment by dizzzz** [comment]
@joewiz yes it is
